motion/thruster_interface_auv_cpp/src/thruster_interface_auv_driver.cpp

```diff
@@ -0,0 +1,138 @@
+#include "thruster_interface_auv_driver.hpp"
+
+ThrusterInterfaceAUVDriver::ThrusterInterfaceAUVDriver(){}
+
+ThrusterInterfaceAUVDriver::ThrusterInterfaceAUVDriver(
+    int I2C_BUS,
+    int PICO_I2C_ADDRESS,
+    double SYSTEM_OPERATIONAL_VOLTAGE,
+    const std::vector<int>& THRUSTER_MAPPING,
+    const std::vector<int>& THRUSTER_DIRECTION,
+    const std::vector<int>& THRUSTER_PWM_OFFSET,
+    const std::vector<int>& PWM_MIN,
+    const std::vector<int>& PWM_MAX,
+    const std::map<int, std::map<std::string, std::vector<double>>>& COEFFS
+) : I2C_BUS(I2C_BUS),
+    PICO_I2C_ADDRESS(PICO_I2C_ADDRESS),
+    //SYSTEM_OPERATIONAL_VOLTAGE(SYSTEM_OPERATIONAL_VOLTAGE), done after in the if-else
+    THRUSTER_MAPPING(THRUSTER_MAPPING),
+    THRUSTER_DIRECTION(THRUSTER_DIRECTION),
+    THRUSTER_PWM_OFFSET(THRUSTER_PWM_OFFSET),
+    PWM_MIN(PWM_MIN),
+    PWM_MAX(PWM_MAX),
+    COEFFS(COEFFS)
+{
+
+    // Convert SYSTEM_OPERATIONAL_VOLTAGE passed as argument to assign the integer field variable [one is double, field var is int]
+    if (SYSTEM_OPERATIONAL_VOLTAGE < 11.0) {
+        this->SYSTEM_OPERATIONAL_VOLTAGE = 10;
+    } else if (SYSTEM_OPERATIONAL_VOLTAGE < 13.0) {
+        this->SYSTEM_OPERATIONAL_VOLTAGE = 12;
+    } else if (SYSTEM_OPERATIONAL_VOLTAGE < 15.0) {
+        this->SYSTEM_OPERATIONAL_VOLTAGE = 14;
+    } else if (SYSTEM_OPERATIONAL_VOLTAGE < 17.0) {
+        this->SYSTEM_OPERATIONAL_VOLTAGE = 16;
+    } else if (SYSTEM_OPERATIONAL_VOLTAGE < 19.0) {
+        this->SYSTEM_OPERATIONAL_VOLTAGE = 18;
+    } else {
+        this->SYSTEM_OPERATIONAL_VOLTAGE = 20;
+    }
```

Comment:
completely removed this part from the code since we discovered there's no handling on the SYSTEM OP VOLTAGE. We dont read it from any topic since we dont measure it, so this piece becomes an unuseful piece of trash. Now we dont pass SYSTEM OP VOLTAGE anymore since it's 16V and not something else. ALWAYS. This mean we dont need anymore the map containing the pair of coeffs for any of them, just the 16V ones. Left all the coeffs in the yaml file for future uses, removed them from the code for saving resources

Reply:
Would it make sense to subscribe to a voltage topic and do this lookup for every thrust output then?

Reply:
it would make sense if we had a way to measure it but we don't. And voltage always stays around 16V, maybe it's 15.5 but it rounds up to 16 so anyway would result in using the 16V values.

Reply:
Fair enough, then there's not much you can do! However, not having a battery voltage measurement available is major haram, might want to whip electronics/embedded to get that up and running. Speaking from experience, if you do not have an alarm system set up to kill everything when the battery voltage dips too low for too long, you WILL destroy batteries :joy:

Reply:
('im totally not sure about this @Andeshog correct me) but we should have a BMS, it's just that's it's purely enclosed in his purpose and doesnt talk with the other systems and so we dont have a "digital" access on it that we could use for this purpose.